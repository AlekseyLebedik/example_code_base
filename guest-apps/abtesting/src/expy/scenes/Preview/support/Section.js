import React from 'react';
import PropTypes from 'prop-types';
import Typography from '@material-ui/core/Typography';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  section: {
    borderTop: '1px solid #E6E6E6',
    marginTop: theme.spacing(3),
    paddingTop: theme.spacing(3),
  },
  titleContainer: {
    marginBottom: '0.5rem',
  },
  title: {
    fontWeight: 'bold',
  },
  description: {
    overflowWrap: 'break-word',
  },
}));

const Section = ({ title, description, children }) => {
  const classes = useStyles();

  if (!description && !children) return null;

  return (
    <div className={classes.section}>
      <div className={classes.titleContainer}>
        <Typography className={classes.title} component="p">
          {title}
        </Typography>
      </div>
      <div className={classes.description}>
        {description ? <p>{description}</p> : children}
      </div>
    </div>
  );
};

Section.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  children: PropTypes.node,
};

Section.defaultProps = {
  title: '',
  description: '',
  children: undefined,
};

export default Section;
