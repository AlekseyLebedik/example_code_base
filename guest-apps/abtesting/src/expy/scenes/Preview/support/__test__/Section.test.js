import React from 'react';
import { shallow } from 'enzyme';

import Section from '../Section';

describe('ABTesting Design - Section', () => {
  it('renders section snapshot', () => {
    expect(shallow(<Section />)).toMatchSnapshot();
  });
});
