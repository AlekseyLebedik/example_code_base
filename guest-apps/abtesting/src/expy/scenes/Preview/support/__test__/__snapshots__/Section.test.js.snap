// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ABTesting Design - Section renders section snapshot 1`] = `""`;
